tests/formats/cif.cpp

```diff
@@ -0,0 +1,107 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include <streambuf>
+#include <fstream>
+
+#include "catch.hpp"
+#include "helpers.hpp"
+#include "chemfiles.hpp"
+using namespace chemfiles;
+
+TEST_CASE("Read files in CIF format") {
+    SECTION("Read single step") {
+        // This is how I imagine most people will resolve the conflict between
+        // CIF files and mmCIF files.
+        auto file = Trajectory("data/cif/1544173.cif", 'r', "CIF");
+        Frame frame = file.read();
+
+        CHECK(frame.size() == 100);
+	CHECK(frame.get("name")->as_string() == "1544173");
+
+        // Check reading positions
+        auto positions = frame.positions();
+        CHECK(approx_eq(positions[0], Vector3D(-0.428, 5.427, 11.536), 1e-3));
+        CHECK(approx_eq(positions[1], Vector3D(4.634, 11.178, 0.405), 1e-3));
+	CHECK(frame[0].name() == "O1");
+	CHECK(frame[1].name() == "O1");
+
+        // Check the unit cell
+        auto cell = frame.cell();
+        CHECK(approx_eq(cell.a(), 5.9170, 1e-4));
+        CHECK(approx_eq(cell.b(), 11.5030, 1e-4));
+        CHECK(approx_eq(cell.c(), 12.0635, 1e-4));
+        CHECK(approx_eq(cell.alpha(), 90., 1e-4));
+        CHECK(approx_eq(cell.beta(), 98.153, 1e-4));
+        CHECK(approx_eq(cell.gamma(), 90., 1e-4));
+    }
+
+    SECTION("Check nsteps and read frame by frame") {
+        auto file = Trajectory("data/cif/Zr-UiO-66-pressure.cif", 'r', "CIF");
+        CHECK(file.nsteps() == 3);
+
+        Frame frame = file.read();
+        CHECK(approx_eq(frame.cell().a(), 20.721205, 1e-4));
+        frame = file.read();
+        CHECK(approx_eq(frame.cell().a(), 20.561233, 1e-4));
+        frame = file.read();
+        CHECK(approx_eq(frame.cell().a(), 20.415146, 1e-4));
```

Comment:
Could you add checks that atom names, properties & position are read as expected?